dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/generate/ServiceImplementationGenerator.java

```diff
@@ -99,9 +103,21 @@ private MethodSpec clientImpl(EndpointDefinition def) {
 
         def.arguments().forEach(arg -> generateParam(arg).ifPresent(methodBuilder::addCode));
 
-        methodBuilder.returns(def.returns());
+        methodBuilder.returns(def.returns().returnType());
+
+        boolean isAsync = def.returns().asyncInnerType().isPresent();
 
-        methodBuilder.addCode("throw new $T();", UnsupportedOperationException.class);
+        String executeCode = isAsync
+                ? "$L.clients().call($L, $L.build" + "(), $L);"
```

Comment:
```suggestion
                ? "$L.clients().call($L, $L.build(), $L);"
```

Reply:
Dang it, why does this not get autoformatted